src/main/java/org/opensearch/security/dlic/rest/api/SecurityConfigApiAction.java

```diff
@@ -73,28 +74,37 @@ protected CType getConfigType() {
     }
 
     private void securityConfigApiActionRequestHandlers(RequestHandler.RequestHandlersBuilder requestHandlersBuilder) {
-        requestHandlersBuilder.onChangeRequest(
-            Method.PUT,
-            request -> withAllowedEndpoint(request).map(this::withConfigEntityNameOnly).map(ignore -> processPutRequest(request))
-        )
-            .onChangeRequest(Method.PATCH, request -> withAllowedEndpoint(request).map(this::processPatchRequest))
+        requestHandlersBuilder.withAccessHandler(this::accessHandler)
+            .verifyAccessForAllMethods()
+            .onChangeRequest(Method.PUT, request -> withConfigEntityNameOnly(request).map(this::processPutRequest))
+            .onChangeRequest(Method.PATCH, this::processPatchRequest)
             .override(Method.DELETE, methodNotImplementedHandler)
             .override(Method.POST, methodNotImplementedHandler);
     }
 
-    ValidationResult<RestRequest> withAllowedEndpoint(final RestRequest request) {
-        if (!allowPutOrPatch) {
-            return ValidationResult.error(RestStatus.NOT_IMPLEMENTED, methodNotImplementedMessage(request.method()));
+    boolean accessHandler(final RestRequest request) {
+        switch (request.method()) {
+            case PATCH:
+            case PUT:
+                if (!allowPutOrPatch && !restApiAdminEnabled) {
+                    return false;
+                } else if (allowPutOrPatch && !restApiAdminEnabled) {
+                    return true;
+                } else {
+                    return securityApiDependencies.restApiAdminPrivilegesEvaluator()
+                        .isCurrentUserAdminFor(endpoint, SECURITY_CONFIG_UPDATE);
+                }
+            default:
+                return true;
         }
-        return ValidationResult.success(request);
     }
 
-    ValidationResult<String> withConfigEntityNameOnly(final RestRequest request) {
+    ValidationResult<RestRequest> withConfigEntityNameOnly(final RestRequest request) {
         final var name = nameParam(request);
         if (!"config".equals(name)) {
```

Comment:
`config` is the only allowed value for this parameter? If that's the case then why do we continue to support `name` as a path param?

Would anyone ever use this endpoint with `internalusers` or `roles` or another config type?

Reply:
> Is `config` the only allowed value for this parameter? If that's the case then why do we continue to support `name` as a path param?
> 
> Would anyone ever use this endpoint with `internalusers` or `roles` or another config type?

Agree. WDYT if I set `config` in this case and other places a part of URL:
instead
`new Route(Method.PUT, "/securityconfig/{name}")`
use 
`new Route(Method.PUT, "/securityconfig/config")`

Reply:
Hi @cwperks, addressed your comment in https://github.com/opensearch-project/security/pull/3359

Reply:
TY @willyborankin , I agree that we should remove the param altogether. I will look at the PR to remove the param ASAP.

Reply:
Rebased with main.